woocommerce/woocommerce
# Cache order year_months in options for performance.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Caches the slow query which calculates year-months to populate the dates filter on order admin screen:

<img width="1232" alt="Screenshot 2024-07-29 at 3 24 50 PM" src="https://github.com/user-attachments/assets/57685bb7-a722-466d-99a7-57c1013ac59e">

We cache the results in wp options table, since these results won't be invalidated soon. We validated the cache again every day since new day might also be a new month. So we update the cache to add this month if needed. Additionally, there might be bulk orders imported for previous days, so in this case, the cache will be invalid only till day is changed.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Make sure HPOS is authoritative.
1. Start with a new site without any orders. Go to the order admin page and verify that current year month is still displayed.
3. In a shop with existing orders (if you only have a new shop, add some orders and manually change the date to previous months), switch to this branch, go to the order admin page, refresh it, and check that the orders are populated as expected. 



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
